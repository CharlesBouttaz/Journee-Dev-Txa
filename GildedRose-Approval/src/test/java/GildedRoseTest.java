import java.util.ArrayList;
import java.util.List;

import org.approvaltests.Approvals;
import org.junit.Test;


public class GildedRoseTest {

	@Test
	public void goldenMasterShouldBeTheSame() throws Exception {
        final GildedRose gildedRose = new GildedRose();
        gildedRose.createItems();

        final List<String> res = new ArrayList<String>();

        for (int i = 0; i < 20; i++) {
            gildedRose.updateQuality();
            res.add(gildedRose.getItems().toString());
        }

        Approvals.verifyAll("Items", res.toArray());
    }

	@Test
	public void externalCallTest() throws Exception {
        final GildedRose gildedRose = new GildedRose();
        gildedRose.createItems();

        final List<String> res = new ArrayList<String>();

        for (int i = 0; i < 3; i++) {
            gildedRose.updateQuality();
            res.add(gildedRose.externalCallParams);
        }

        Approvals.verifyAll("Ordered Items", res.toArray());
    }

    public void parameterisedTests() {
        //LegacyApprovals.LockDown(object, "method", param1Range, param2range);
        // See https://www.youtube.com/watch?v=EJ1tRFEOkcw
    }

    public void binaryCompareTest() {
        // PDF
//        Approvals.verify(byte[])
//        Approvals.verify(inputStream)
    }
}
