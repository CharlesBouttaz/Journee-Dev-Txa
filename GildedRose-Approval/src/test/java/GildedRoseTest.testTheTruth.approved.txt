Items[0] = [Item{name='+5 Dexterity Vest', sellIn=9, quality=19}, Item{name='Aged Brie', sellIn=1, quality=1}, Item{name='Elixir of the Mongoose', sellIn=4, quality=6}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=14, quality=21}, Item{name='Conjured Mana Cake', sellIn=2, quality=5}]
Items[1] = [Item{name='+5 Dexterity Vest', sellIn=8, quality=18}, Item{name='Aged Brie', sellIn=0, quality=2}, Item{name='Elixir of the Mongoose', sellIn=3, quality=5}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=13, quality=22}, Item{name='Conjured Mana Cake', sellIn=1, quality=4}]
Items[2] = [Item{name='+5 Dexterity Vest', sellIn=7, quality=17}, Item{name='Aged Brie', sellIn=-1, quality=4}, Item{name='Elixir of the Mongoose', sellIn=2, quality=4}, Item{name='Sulfuras, Hand of Ragnaros', sellIn=0, quality=80}, Item{name='Backstage passes to a TAFKAL80ETC concert', sellIn=12, quality=23}, Item{name='Conjured Mana Cake', sellIn=0, quality=3}]
